### DISCUSSION TITLE
Examples to Understanding 3.3.4 Error Prevention?

### DISCUSSION THREAD

**Discussion by Ignacio-de-Loyola** [w3c/wcag#3249] [Q&A]
![image](https://github.com/w3c/wcag/assets/80868256/3e44e668-dbc4-4102-b5b8-ce4e47131e99)
Is it against accessibility if there is no cancel button(e.g. X) in a dialog without cancel function as in the example above? The only way to return to the previous step is to use 'Back button' after switching screens. 

In a practical case, the two buttons on the bottom ('No, Thaks' and 'Payment') are the means to move on to the next step in the dialog below. In this case, the only way the user can cancel is by using the ESC key. In this case, is it against accessibility because it cannot be reversed?
![image](https://github.com/w3c/wcag/assets/80868256/80b7d60f-7a49-4be8-94e7-5a78a0eefe3a)

**Comment by patrickhlauke** [comment]
converted to a discussion, as (very high-level) question is not about the spec itself.

**Comment by patrickhlauke** [comment]
> The only way to return to the previous step is to use 'Back button' after switching screens

if that has been implemented correctly, that's possibly a reasonable approach. however, in most cases (such as that bootstrap-looking modal you have there) the back button doesn't actually close the modal as such, but navigates to the previous page in the browser's history, so it's not equivalent to an actual close control.

and just relying on the user being able to press `Esc` relies on the fact that they have a keyboard, which may not always be the case (particularly on mobile/tablet scenarios), so probably not a good approach either.

**Comment by Ignacio-de-Loyola** [comment]
@patrickhlauke Is what you said 'just a bad approach(UX), not against accessibility' correct?
